In [1]:
import pandas as pd
import seaborn as sns

In [9]:
tsa=pd.read_csv('tsa_claims1.csv')
tsa.head()

C:\Users\2518533\AppData\Local\Temp\ipykernel_8992\882036362.py:1: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.
  tsa=pd.read_csv('tsa_claims1.csv')


Claim Number Date Received    Incident Date Airport Code  \
0       0909802M      4-Jan-02  12/12/2002 0:00          EWR   
1       0202445M      4-Feb-02  11/26/2003 0:00          STL   
2       0909816M      7-Feb-02    1/6/2003 0:00          MIA   
3  2005032379513     18-Feb-02    2/5/2005 0:00          MCO   
4  2005032479636     18-Feb-02   2/17/2005 0:00          CAK   

                      Airport Name                   Airline Name  \
0     Newark International Airport           Continental Airlines   
1  Lambert St. Louis International              American Airlines   
2      Miami International Airport              American Airlines   
3    Orlando International Airport                   Delta (Song)   
4            Akron-Canton Regional  Air Tran Airlines(do not use)   

                Claim Type       Claim Site  \
0          Property Damage       Checkpoint   
1          Property Damage  Checked Baggage   
2          Property Damage       Checkpoint   
3          Property Damage       Checkpoint   
4  Passenger Property Loss  Checked Baggage   

                                         Item  Claim Amount    Status  \
0                                       Other        350.00  Approved   
1                                 Cell Phones        278.88   Settled   
2   Luggage (all types including footlockers)         50.00  Approved   
3  Baby - Strollers; car seats; playpen; etc.         84.79  Approved   
4                                       Other        459.01   Settled   

   Close Amount      Disposition  
0        350.00  Approve in Full  
1        227.92           Settle  
2         50.00  Approve in Full  
3         84.79  Approve in Full  
4        227.00           Settle

In [21]:
tsa.duplicated(subset=['Claim Amount','Item'], keep='first')

0        False
1        False
2        False
3        False
4        False
         ...  
94843     True
94844    False
94845     True
94846     True
94847    False
Length: 94848, dtype: bool

In [40]:
tsa.drop_duplicates(subset=['Claim Amount','Item', 'Claim Number'], keep='first')

Claim Number Date Received     Incident Date Airport Code  \
0           0909802M      4-Jan-02   12/12/2002 0:00          EWR   
1           0202445M      4-Feb-02   11/26/2003 0:00          STL   
2           0909816M      7-Feb-02     1/6/2003 0:00          MIA   
3      2005032379513     18-Feb-02     2/5/2005 0:00          MCO   
4      2005032479636     18-Feb-02    2/17/2005 0:00          CAK   
...              ...           ...               ...          ...   
94843  2010010567195     31-Dec-09   12/14/2009 0:00          EWR   
94844  2010010667196     31-Dec-09   12/17/2009 7:55          BWI   
94845  2010010667198     31-Dec-09   12/15/2009 0:00          SMF   
94846  2010010667199     31-Dec-09  12/16/2009 10:00          ENV   
94847  2009122866807     31-Dec-09   11/21/2009 9:20          JFK   

                                      Airport Name  \
0                     Newark International Airport   
1                  Lambert St. Louis International   
2                      Miami International Airport   
3                    Orlando International Airport   
4                            Akron-Canton Regional   
...                                            ...   
94843                 Newark International Airport   
94844  Baltimore/Washington Intl Thurgood Marshall   
94845             Sacramento International Airport   
94846                   Wendover Municipal Airport   
94847                John F. Kennedy International   

                        Airline Name               Claim Type  \
0               Continental Airlines          Property Damage   
1                  American Airlines          Property Damage   
2                  American Airlines          Property Damage   
3                       Delta (Song)          Property Damage   
4      Air Tran Airlines(do not use)  Passenger Property Loss   
...                              ...                      ...   
94843           Continental Airlines          Property Damage   
94844             Southwest Airlines  Passenger Property Loss   
94845             Southwest Airlines  Passenger Property Loss   
94846                   Xtra Airways  Passenger Property Loss   
94847              American Airlines          Property Damage   

            Claim Site                                               Item  \
0           Checkpoint                                              Other   
1      Checked Baggage                                        Cell Phones   
2           Checkpoint          Luggage (all types including footlockers)   
3           Checkpoint         Baby - Strollers; car seats; playpen; etc.   
4      Checked Baggage                                              Other   
...                ...                                                ...   
94843  Checked Baggage          Luggage (all types including footlockers)   
94844  Checked Baggage  Cell Phones (smartphone type-blackberry; iphon...   
94845  Checked Baggage                                              Locks   
94846  Checked Baggage                                              Locks   
94847       Checkpoint                                  Computer - Laptop   

       Claim Amount    Status  Close Amount      Disposition  
0            350.00  Approved        350.00  Approve in Full  
1            278.88   Settled        227.92           Settle  
2             50.00  Approved         50.00  Approve in Full  
3             84.79  Approved         84.79  Approve in Full  
4            459.01   Settled        227.00           Settle  
...             ...       ...           ...              ...  
94843        175.90    Denied          0.00             Deny  
94844        695.00    Denied          0.00             Deny  
94845         25.50    Denied          0.00             Deny  
94846         20.00    Denied          0.00             Deny  
94847       1228.95   Settled        826.72           Settle  

[94848 rows x 13 columns]

In [30]:
tsa.dropna()

Claim Number Date Received     Incident Date Airport Code  \
0           0909802M      4-Jan-02   12/12/2002 0:00          EWR   
1           0202445M      4-Feb-02   11/26/2003 0:00          STL   
2           0909816M      7-Feb-02     1/6/2003 0:00          MIA   
3      2005032379513     18-Feb-02     2/5/2005 0:00          MCO   
4      2005032479636     18-Feb-02    2/17/2005 0:00          CAK   
...              ...           ...               ...          ...   
94843  2010010567195     31-Dec-09   12/14/2009 0:00          EWR   
94844  2010010667196     31-Dec-09   12/17/2009 7:55          BWI   
94845  2010010667198     31-Dec-09   12/15/2009 0:00          SMF   
94846  2010010667199     31-Dec-09  12/16/2009 10:00          ENV   
94847  2009122866807     31-Dec-09   11/21/2009 9:20          JFK   

                                      Airport Name  \
0                     Newark International Airport   
1                  Lambert St. Louis International   
2                      Miami International Airport   
3                    Orlando International Airport   
4                            Akron-Canton Regional   
...                                            ...   
94843                 Newark International Airport   
94844  Baltimore/Washington Intl Thurgood Marshall   
94845             Sacramento International Airport   
94846                   Wendover Municipal Airport   
94847                John F. Kennedy International   

                        Airline Name               Claim Type  \
0               Continental Airlines          Property Damage   
1                  American Airlines          Property Damage   
2                  American Airlines          Property Damage   
3                       Delta (Song)          Property Damage   
4      Air Tran Airlines(do not use)  Passenger Property Loss   
...                              ...                      ...   
94843           Continental Airlines          Property Damage   
94844             Southwest Airlines  Passenger Property Loss   
94845             Southwest Airlines  Passenger Property Loss   
94846                   Xtra Airways  Passenger Property Loss   
94847              American Airlines          Property Damage   

            Claim Site                                               Item  \
0           Checkpoint                                              Other   
1      Checked Baggage                                        Cell Phones   
2           Checkpoint          Luggage (all types including footlockers)   
3           Checkpoint         Baby - Strollers; car seats; playpen; etc.   
4      Checked Baggage                                              Other   
...                ...                                                ...   
94843  Checked Baggage          Luggage (all types including footlockers)   
94844  Checked Baggage  Cell Phones (smartphone type-blackberry; iphon...   
94845  Checked Baggage                                              Locks   
94846  Checked Baggage                                              Locks   
94847       Checkpoint                                  Computer - Laptop   

       Claim Amount    Status  Close Amount      Disposition  
0            350.00  Approved        350.00  Approve in Full  
1            278.88   Settled        227.92           Settle  
2             50.00  Approved         50.00  Approve in Full  
3             84.79  Approved         84.79  Approve in Full  
4            459.01   Settled        227.00           Settle  
...             ...       ...           ...              ...  
94843        175.90    Denied          0.00             Deny  
94844        695.00    Denied          0.00             Deny  
94845         25.50    Denied          0.00             Deny  
94846         20.00    Denied          0.00             Deny  
94847       1228.95   Settled        826.72           Settle  

[94848 rows x 13 columns]

In [62]:
grop=tsa.groupby(by=['Claim Type']).count()
grop

Claim Number  Date Received  Incident Date  \
Claim Type                                                            
Employee Loss (MPCECA)            308            308            308   
Motor Vehicle                       6              6              6   
Passenger Property Loss         60265          60265          60265   
Passenger Theft                   331            331            331   
Personal Injury                   208            208            208   
Property Damage                 33730          33730          33730   

                         Airport Code  Airport Name  Airline Name  Claim Site  \
Claim Type                                                                      
Employee Loss (MPCECA)            308           308           308         308   
Motor Vehicle                       6             6             6           6   
Passenger Property Loss         60265         60265         60265       60265   
Passenger Theft                   331           331           331         331   
Personal Injury                   208           208           208         208   
Property Damage                 33730         33730         33730       33730   

                          Item  Claim Amount  Status  Close Amount  \
Claim Type                                                           
Employee Loss (MPCECA)     308           308     308           308   
Motor Vehicle                6             6       6             6   
Passenger Property Loss  60265         60265   60265         60265   
Passenger Theft            331           331     331           331   
Personal Injury            208           208     208           208   
Property Damage          33730         33730   33730         33730   

                         Disposition  
Claim Type                            
Employee Loss (MPCECA)           308  
Motor Vehicle                      6  
Passenger Property Loss        60265  
Passenger Theft                  331  
Personal Injury                  208  
Property Damage                33730

# •	What is the most common type of insurance claim?
Passenger Property loss

In [67]:
site=tsa.groupby(by=['Claim Site']).count()
site

Claim Number  Date Received  Incident Date  Airport Code  \
Claim Site                                                                  
Checked Baggage         80553          80553          80553         80553   
Checkpoint              14013          14013          14013         14013   
Motor Vehicle               2              2              2             2   
Other                     280            280            280           280   

                 Airport Name  Airline Name  Claim Type   Item  Claim Amount  \
Claim Site                                                                     
Checked Baggage         80553         80553       80553  80553         80553   
Checkpoint              14013         14013       14013  14013         14013   
Motor Vehicle               2             2           2      2             2   
Other                     280           280         280    280           280   

                 Status  Close Amount  Disposition  
Claim Site                                          
Checked Baggage   80553         80553        80553  
Checkpoint        14013         14013        14013  
Motor Vehicle         2             2            2  
Other               280           280          280

# •	Which claim site within the airport are claims most commonly filed for?
Checked Baggage

In [76]:
claim=tsa.groupby(by=['Claim Site','Claim Type']).count()
claim

Claim Number  Date Received  \
Claim Site      Claim Type                                             
Checked Baggage Employee Loss (MPCECA)            275            275   
                Motor Vehicle                       4              4   
                Passenger Property Loss         53364          53364   
                Passenger Theft                   294            294   
                Personal Injury                    89             89   
                Property Damage                 26527          26527   
Checkpoint      Employee Loss (MPCECA)             32             32   
                Motor Vehicle                       1              1   
                Passenger Property Loss          6740           6740   
                Passenger Theft                    36             36   
                Personal Injury                   112            112   
                Property Damage                  7092           7092   
Motor Vehicle   Passenger Property Loss             2              2   
Other           Employee Loss (MPCECA)              1              1   
                Motor Vehicle                       1              1   
                Passenger Property Loss           159            159   
                Passenger Theft                     1              1   
                Personal Injury                     7              7   
                Property Damage                   111            111   

                                         Incident Date  Airport Code  \
Claim Site      Claim Type                                             
Checked Baggage Employee Loss (MPCECA)             275           275   
                Motor Vehicle                        4             4   
                Passenger Property Loss          53364         53364   
                Passenger Theft                    294           294   
                Personal Injury                     89            89   
                Property Damage                  26527         26527   
Checkpoint      Employee Loss (MPCECA)              32            32   
                Motor Vehicle                        1             1   
                Passenger Property Loss           6740          6740   
                Passenger Theft                     36            36   
                Personal Injury                    112           112   
                Property Damage                   7092          7092   
Motor Vehicle   Passenger Property Loss              2             2   
Other           Employee Loss (MPCECA)               1             1   
                Motor Vehicle                        1             1   
                Passenger Property Loss            159           159   
                Passenger Theft                      1             1   
                Personal Injury                      7             7   
                Property Damage                    111           111   

                                         Airport Name  Airline Name   Item  \
Claim Site      Claim Type                                                   
Checked Baggage Employee Loss (MPCECA)            275           275    275   
                Motor Vehicle                       4             4      4   
                Passenger Property Loss         53364         53364  53364   
                Passenger Theft                   294           294    294   
                Personal Injury                    89            89     89   
                Property Damage                 26527         26527  26527   
Checkpoint      Employee Loss (MPCECA)             32            32     32   
                Motor Vehicle                       1             1      1   
                Passenger Property Loss          6740          6740   6740   
                Passenger Theft                    36            36     36   
                Personal Injury                   112           112    112   
 

# •	What type of claim is made most at each claim site?
Checked Baggage: Passenger Property Loss	
Motor Vehicle: Passenger Property Loss	
checkpoint: Property Damage	

In [95]:
claimamt=tsa['Claim Amount'].mode()
claimamt

0    100.0
Name: Claim Amount, dtype: float64

# •	What is the typical claim amount?
100,0

In [102]:
# Count total claims
total_claims = tsa.shape[0]

approved_claims = tsa[tsa['Status'] == 'Approved'].shape[0]

approval_rate = (approved_claims / total_claims) * 100 if total_claims > 0 else 0

approval_rate

24.425396423751685

# •	What is the overall claim approval rate for the entire U.S.?

24.43%